probe-rs/src/core/registers.rs

```diff
@@ -21,11 +21,11 @@ pub enum RegisterDataType {
 /// This denotes the purpose of a register (e.g. `return address`),
 /// while the [`CoreRegister::name`] will contain the architecture specific label of the register.
 #[derive(Debug, Clone, Copy, PartialEq)]
-pub(crate) enum RegisterRole {
-    /// Argument registers like "A0", "a1", "r2", etc. (uses architecture specific names)
+pub enum RegisterRole {
+    /// Function Argument registers like "A0", "a1", etc. (uses architecture specific names)
     Argument(&'static str),
-    /// Some argument registers can also be used as return value registers.
-    ArgumentAndResult(&'static str),
+    /// Function Return value registers like "R0", "r1", etc. (uses architecture specific names)
+    Return(&'static str),
```

Comment:
Why do return registers have names, but e.g. the floating point registers have no name? They seem to be named as well.

Reply:
Good catch. I will add it :)

Reply:
You are right (as usual 😀 ) ... it is still possible to add multiple definitions for a single register, just not necessary to do so in most cases (keeps the duplication to a minimum). This example with multiple datatype FP registers would be the exception, but until now, I guess nobody has seen the need to create them.